plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryPagesStore.java

```diff
@@ -117,7 +117,10 @@ public synchronized List<Page> getPages(
                 done = true;
             }
             // Append missing columns with null values. This situation happens when a new column is added without additional insert.
-            for (int j = page.getChannelCount(); j < columnIndexes.length; j++) {
+            for (int j = 0; j < columnIndexes.length; j++) {
+                if (columnIndexes[j] < page.getChannelCount()) {
```

Comment:
Is this related to dropping columns, or a bug fix? Should this be in a separate commit?